File: common/src/main/java/cc/turtl/chiselmon/system/alert/action/GlowAction.java
package cc.turtl.chiselmon.system.alert.action;

import cc.turtl.chiselmon.system.alert.AlertContext;
import static cc.turtl.chiselmon.client.util.PokemonEntityExtensionsKt.*;

public class GlowAction implements AlertAction {
    @Override
    public void execute(AlertContext ctx) {
        if (ctx.shouldHighlight()) {
            int color = ctx.highlightFilter().rgb();
            addGlow(ctx.entity(), color);
            highlightNickname(ctx.entity(), color);
        }
    }
}

File: common/src/main/kotlin/cc/turtl/chiselmon/client/system/alert/action/GlowAction.kt
package cc.turtl.chiselmon.client.system.alert.action

import cc.turtl.chiselmon.client.util.addGlow
import cc.turtl.chiselmon.client.util.highlightNickname
import cc.turtl.chiselmon.client.system.alert.AlertContext

class GlowAction : AlertAction {
    override fun execute(ctx: AlertContext) {
        val filter = ctx.highlightFilter ?: return
        if (!ctx.shouldHighlight) return
        ctx.entity.addGlow(filter.rgb)
        ctx.entity.highlightNickname(filter.rgb)
    }
}
